app/main/helpers/search_helpers.py

```diff
@@ -4,8 +4,9 @@
 from werkzeug.datastructures import MultiDict
 
 
-def get_lot_from_request(request):
-    return request.args.get('lot', None)
+def get_lot_from_request(request, all_lots):
+    lot = request.args.get('lot', None)
+    return lot if lot in all_lots or not all_lots else None
```

Comment:
Damn, yeah. I had the not check first, and then moved it because I didn't like starting with a not. My bad. Good spot.